assets/js/spellchecker.js

```diff
@@ -0,0 +1,254 @@
+    });
+    spellerArray.sort(function (a, b) {
+        return getLangByCode(a[0]).localeCompare(getLangByCode(b[0]));
+    });
+
+    for(var i = 0; i < spellerArray.length; i++) {
+        lang = spellerArray[i][0];
+        $('#primarySpellCheckerMode').append($('<option></option>').val(lang).text(getLangByCode(lang)));
+    }
+
+    restoreChoices('spellerchecker');
+}
+
+function check() {
+    if(currentSpellCheckerRequest) {
+        currentSpellCheckerRequest.abort();
+    }
+    $('#spellCheckerInput').addClass('spellCheckerVisible');
+    $('#spellCheckerInput').html($('#spellCheckerInput').html().replace(/br/g, '\n')
+        .replace(/&nbsp;/g, ' '));
+    var words = $.trim($('#spellCheckerInput').text());
+    var splitWords = words.split(' ');
+    var content = {};
+    $('#spellCheckerInput').html('');
+    currentSpellCheckerRequest = callApy({
+        data: {
+            'q': words,
+            'lang': $('#primarySpellCheckerMode').val()
+        },
+        success: function (data) {
+            var originalWordsIndex = 0;
+            for(var tokenIndex = 0; tokenIndex < data.length; tokenIndex++) {
+                if(data[tokenIndex]['known'] === true) { // eslint-disable-line dot-notation
+                    $('#spellcheckerInput').html($('#spellcheckerInput').html() + ' ' + splitWords[originalWordsIndex]);
+                    originalWordsIndex++;
+                    continue;
+                }
+                $('#spellcheckerInput').html($('#spellcheckerInput').html() + ' <span class="spellError" id=' +
+                    splitWords[originalWordsIndex] + '>' + splitWords[originalWordsIndex] + '</span>');
+                content[splitWords[originalWordsIndex]] = '<div class="list-group">';
+                for(var sugg = 0; sugg < data[tokenIndex]['sugg'].length; sugg++) { // eslint-disable-line dot-notation
+                    content[splitWords[originalWordsIndex]] += '<a href="#" class="list-group-item">' +
+                        data[tokenIndex]['sugg'][sugg][0] + '</a>'; // eslint-disable-line dot-notation
+                    content[splitWords[originalWordsIndex]] += '</div>';
+                }
+                $('.spellError').each(function () {
+                    var currentTokenId = this.id;
+                    $(this).popover({animation: false, placement: 'bottom', trigger: 'manual', html: true,
+                        content: content[currentTokenId]});
+                });
+                originalWordsIndex++;
+            }
+        },
+        error: handleSpellCheckerErrorResponse,
+        complete: function () {
+            ajaxComplete();
+            currentSpellCheckerRequest = undefined;
+        }
+    }, '/speller', true);
+}
+
+function handleSpellCheckerErrorResponse(jqXHR) {
+    spellCheckerNotAvailable(jqXHR.responseJSON);
+}
+
+function spellCheckerNotAvailable(data) {
+    $('#spellCheckerInput').append($('<div></div>').text(' '));
```

Comment:
What's the point of `.text(' ')`?

Reply:
It seems to prevent a new line from appearing before the message.